packages/blade/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,463 @@
+        }
+      } else {
+        newY = predictedY;
+      }
+
+      const isPosAtUpperSnapPoint = newY >= upperSnapPoint;
+
+      if (isContentDragging) {
+        if (isPosAtUpperSnapPoint) {
+          newY = upperSnapPoint;
+        }
+
+        // keep the newY at upper snap point
+        // until the scrollable content is not at top
+        // and previously saved Y position is greater than or equal to upper snap point
+        // Note: how using newY won't work here since we need the previous value of the newY
+        // since we always keep updating the newY,
+        // this is cruicial in making the scroll feel natural
+        const isContentScrolledAtTop = scrollRef.current && scrollRef.current.scrollTop <= 0;
+        if (lastOffsetY === upperSnapPoint && !isContentScrolledAtTop) {
+          newY = upperSnapPoint;
+        }
+        preventScrollingRef.current = newY < upperSnapPoint;
+      }
+
+      const shouldClose = newY < lowerSnapPoint;
+      if (shouldClose) {
+        setIsDragging(false);
+        close();
+        cancel();
+        return;
+      }
+
+      if (last) {
+        // calculate the nearest snapPoint
+        const [nearest] = computeSnapPointBounds(
+          newY,
+          snapPoints.map((point) => dimensions.height * point) as SnapPoints,
+        );
+
+        // if we stop dragging assign snap to the nearest point
+        if (!active && !tap) {
+          newY = nearest;
+        }
+      }
+
+      setPositionY(newY, !down);
+    },
+    {
+      from: [0, positionY],
+      filterTaps: true,
+      enabled: isOnTopOfStack && _isOpen,
+    },
+  );
+
+  // Here we are preventing the scrolling of the content, until the preventScrollingRef value is true
+  useIsomorphicLayoutEffect(() => {
+    const scrollElement = scrollRef.current;
+    if (!scrollElement) return;
+
+    const preventScrolling = (e: Event) => {
+      if (preventScrollingRef?.current) {
+        e.preventDefault();
+      }
+    };
+
+    // https://www.bram.us/2016/05/02/prevent-overscroll-bounce-in-ios-mobilesafari-pure-css/
+    const preventSafariOverscroll = (e: Event) => {
+      if (scrollElement.scrollTop < 0) {
+        // TODO: figure this out, it doesn't seem to work >iOS12
+        // requestAnimationFrame(() => {
+        //   elem.style.overflow = 'hidden';
+        //   elem.scrollTop = 0;
+        //   elem.style.removeProperty('overflow');
+        // });
+        e.preventDefault();
+      }
+    };
+
+    scrollElement.addEventListener('scroll', preventScrolling);
+    scrollElement.addEventListener('touchmove', preventScrolling);
+    scrollElement.addEventListener('touchstart', preventSafariOverscroll);
+    return () => {
+      scrollElement.removeEventListener('scroll', preventScrolling);
+      scrollElement.removeEventListener('touchmove', preventScrolling);
+      scrollElement.removeEventListener('touchstart', preventSafariOverscroll);
+    };
+  }, [scrollRef.current]);
```

Comment:
Correct, this is wrong. fixed the logic. 

just passing the ref or the empty dep array wont work so passed the isReady flag to the dep. This will ensure that we are done with all the layout calculation when this hook runs. By then the scrollRef.current will be populated. 

```
const isReady = contentHeight > 0
```